### PR TITLE
RavenDB-15071 Replication Hub Certificates - Add Filtering in Studio

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts [hunk 1]
```diff
@@ -43,7 +43,16 @@ class editReplicationHubTask extends viewModelBase {
         saveReplicationAccess: ko.observable<boolean>(false),
         generateCertificate: ko.observable<boolean>(false)
     };
+
+    allReplicationAccessItems = ko.observableArray<replicationAccessHubModel>([]);
+    visibleReplicationAccessItems: KnockoutComputed<Array<replicationAccessHubModel>>;
+    
+    filterText = ko.observable<string>();
     
+    readonly accessItemsBatch = 10;
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts]: I think we want to have 100 here?

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts [hunk 2]
```diff
@@ -139,6 +148,29 @@ class editReplicationHubTask extends viewModelBase {
     }
 
     private initObservables() {
+        this.visibleReplicationAccessItems = ko.pureComputed(() => {
+            let itemsToShow = this.allReplicationAccessItems();
+
+            const filter = this.filterText();
+            if (filter) {
+                const upperFilter = filter.toUpperCase();
+                itemsToShow = itemsToShow.filter(x => x.replicationAccessName().toLowerCase().includes(upperFilter) ||
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts]: can we avoid recomputing filtered items on every load?

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts]: As discussed - changing to 50

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts]: done

**Review by ml054 — APPROVED** [review]

**Pull request merged**